packages/syft/src/syft/service/notification/email_templates.py

```diff
@@ -22,6 +23,88 @@ def email_body(notification: "Notification", context: AuthedServiceContext) -> s
         return ""
 
 
+@serializable(canonical_name="PasswordResetTemplate", version=1)
+class PasswordResetTemplate(EmailTemplate):
+    @staticmethod
+    def email_title(notification: "Notification", context: AuthedServiceContext) -> str:
+        return "Password Reset Requested"
+
+    @staticmethod
+    def email_body(notification: "Notification", context: AuthedServiceContext) -> str:
+        user_service = context.server.get_service("userservice")
+
+        user = user_service.get_by_verify_key(notification.to_user_verify_key)
+        if not user:
+            raise Exception("User not found!")
+
+        user.reset_token = user_service.generate_new_password_reset_token(
+            context.server.settings.pwd_token_config
+        )
+        user.reset_token_date = datetime.now()
+
+        result = user_service.stash.update(
+            credentials=context.credentials, user=user, has_permission=True
+        )
+        if result.is_err():
+            raise Exception("Couldn't update the user password")
+
+        head = """<head>
+            <style>
+                body {
+                    font-family: Arial, sans-serif;
+                    background-color: #f4f4f4;
+                    color: #333;
+                    margin: 0;
+                    padding: 0;
+                }
+                .container {
+                    max-width: 600px;
+                    margin: 50px auto;
+                    background-color: #ffffff;
+                    padding: 20px;
+                    border-radius: 8px;
+                    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
+                }
+                h1 {
+                    color: #333;
+                    text-align: center;
+                }
+                p {
+                    font-size: 16px;
+                    line-height: 1.5;
+                    text-align: center;
+                }
+                .button {
+                    display: block;
+                    width: 200px;
+                    margin: 30px auto;
+                    padding: 10px;
+                    background-color: #007BFF;
+                    color: #fff;
+                    text-align: center;
+                    text-decoration: none;
+                    border-radius: 4px;
+                }
+                .footer {
+                    text-align: center;
+                    font-size: 12px;
+                    color: #aaa;
+                    margin-top: 20px;
+                }
+            </style>
+        </head>"""
+        body = f"""<body>
+            <div class="container">
+                <h1>Password Reset</h1>
+                <p>Hello,</p>
+                <p>We received a request to reset your password. Your new temporary token is:</p>
+                <h1>{user.reset_token}</h1>
```

Comment:
There would be a follow-up action after this where the user needs to call `user.reset_password` with this token and the new password right? Should we make that instruction explicit in this template?